packages/core/src/core/fetch.ts

```diff
@@ -292,25 +311,70 @@ const fetchAndProcessMergeErrors = async (
   try {
     const progressReporters = _.mapValues(
       adapters,
-      (_adapter, adapterName) => createAdapterProgressReporter(adapterName, 'fetch', progressEmitter)
+      (_adapter, accountName) => createAdapterProgressReporter(accountName, 'fetch', progressEmitter)
     )
     const fetchResults = await Promise.all(
       Object.entries(adapters)
-        .map(async ([adapterName, adapter]) => {
+        .map(async ([accountName, adapter]) => {
           const fetchResult = await adapter.fetch({
-            progressReporter: progressReporters[adapterName],
+            progressReporter: progressReporters[accountName],
           })
+          const { updatedConfig, errors } = fetchResult
+          if (fetchResult.elements.length > 0 && accountName !== fetchResult.elements[0]
+            .elemID.adapter) {
+            const transformElemID: TransformFunc = async ({ value }) => {
+              if (isReferenceExpression(value)) {
+                return value.clone(value.elemID.createAdapterReplacedId(accountName))
+              }
+              if (isElement(value) && value.elemID.adapter !== accountName) {
+                const transformed = value.clone(undefined, value.elemID
+                  .createAdapterReplacedId(accountName))
```

Comment:
same nit re line breaks
```suggestion
                const transformed = value.clone(
                  undefined,
                  value.elemID.createAdapterReplacedId(accountName),
                )
```